components/humantask/org.wso2.carbon.humantask/src/main/java/org/wso2/carbon/humantask/core/api/client/TaskOperationsImpl.java

```diff
@@ -1772,4 +1806,41 @@ private void handleException(Exception ex) throws IllegalStateFault, IllegalOper
             throw new IllegalStateFault(ex.getMessage());
         }
     }
+
+    /**
+     * Checks whether the tenant is loaded if not loads the tenant.
+     *
+     * Note: This is required only if tenant configurations
+     * need to be redeployed.
+     */
+    private void loadTenant() {
+
+        String tenantDomain = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+
+        if (MultitenantConstants.SUPER_TENANT_DOMAIN_NAME.equals(tenantDomain)) {
+            return;
+        }
+
+        ConfigurationContextService configurationContextService = getConfigurationContextService();
+        if (configurationContextService == null) {
+            log.warn("ConfigurationContextService is not available cannot attempt to load tenant. " +
+                    "Proceeding without tenant loading for domain " + tenantDomain);
+            return;
+        }
+        Set<String> loadedTenants = TenantAxisUtils.getTenantConfigurationContexts(
+                configurationContextService.getServerConfigContext()).keySet();
+        if (!loadedTenants.contains(tenantDomain)) {
+            if (log.isDebugEnabled()) {
+                log.debug("Tenant: " + tenantDomain + " is not loaded. Therefore attempting to load the tenant.");
+            }
+            TenantAxisUtils.getTenantConfigurationContext(tenantDomain,
+                    configurationContextService.getServerConfigContext());
+        }
+    }
+
+    private ConfigurationContextService getConfigurationContextService() {
+
+        return (ConfigurationContextService)PrivilegedCarbonContext.getThreadLocalCarbonContext()
```

Comment:
This osgi service was not bound at server startup as there was some issues when doing that. The server waits on the startup of the HumanTask service.